### PR TITLE
fix: Insert section at index 0

### PR DESCRIPTION
**Pull request by bruno-garcia**
I'm unsure why it started at index=1 but I'm assuming there was a reason.
All I did for now was to "get the code to work" without really knowing why it should insert at Index 1 and how it should be handled if no`section` exists.

The commented out code:

`Assert.Throws<ArgumentOutOfRangeException>(() => ElfObjectFile.TryRead(stream, out _, out _));` 

Serves as a repro, would be deleted before merging if this is to be merged.

### CODE REVIEW

File: src/LibObjectFile/Elf/ElfReader{TDecoder}.cs [hunk 1]
```diff
@@ -508,7 +508,7 @@ private void VerifyAndFixProgramHeadersAndSections()
             if (hasShadowSections)
             {
                 int shadowCount = 0;
-                uint previousSectionIndex = 1;
+                uint previousSectionIndex = 0;
```
Comment (xoofx) [inline src/LibObjectFile/Elf/ElfReader{TDecoder}.cs]: I will have to double check, but it was 1 originally for the index 0 which is supposed to be the null section.

File: src/LibObjectFile.Tests/Elf/ElfSimpleTests.cs [hunk 2]
```diff
@@ -12,6 +13,20 @@ namespace LibObjectFile.Tests.Elf
...
     public class ElfSimpleTests : ElfTestBase
     {
+        [Test]
+        public void TryReadThrows()
+        {
+            using var stream = File.OpenRead("TestFiles/cmnlib.b00");
+            Assert.False(ElfObjectFile.TryRead(stream, out var elf, out var diagnostics));
+            Assert.NotNull(elf);
+            Assert.AreEqual(
+                "Unable to read entirely program header [3]. Not enough data (size: 32) read at offset 148 from the stream",
```
Comment (xoofx) [inline src/LibObjectFile.Tests/Elf/ElfSimpleTests.cs]: better check directly diagnostics.Count == 1, and DiagnosticId instead of matching strings.

Comment (bruno-garcia) [reply src/LibObjectFile.Tests/Elf/ElfSimpleTests.cs]: Done

File: src/LibObjectFile.Tests/Elf/ElfSimpleTests.cs [hunk 3]
```diff
@@ -12,6 +12,16 @@ namespace LibObjectFile.Tests.Elf
...
     public class ElfSimpleTests : ElfTestBase
     {
+        [Test]
+        public void TryReadThrows()
+        {
+            using var stream = File.OpenRead("TestFiles/cmnlib.b00");
+            Assert.False(ElfObjectFile.TryRead(stream, out var elf, out var diagnostics));
+            Assert.NotNull(elf);
+            Assert.AreEqual(1, diagnostics.Messages.Count);
+            Assert.AreEqual(116, diagnostics.Messages[0].Id);
```
Comment (xoofx) [inline src/LibObjectFile.Tests/Elf/ElfSimpleTests.cs]: Why not using the enum DiagnosticId directly?

**Comment by xoofx** [comment]
I had a look at this issue and made an additional commit 6155fb1e94bb99a641c16abecdd5cbc018175bfe to your branch before merging it.

As it is an invalid ELF, errors were not being recovered correctly in that case.

File: src/LibObjectFile.Tests/Elf/ElfSimpleTests.cs (hunk 3 above)
Comment (bruno-garcia) [reply src/LibObjectFile.Tests/Elf/ElfSimpleTests.cs]: Sorry I missed that

**Pull request merged**